sympy/matrices/common.py

```diff
@@ -790,35 +791,33 @@ def diag(kls, *args, **kwargs):
         diag_entries = defaultdict(int)
         R = C = 0  # keep track of the biggest index seen
         for m in args:
-            if hasattr(m, 'rows') or isinstance(m, list):
-                # in this case, we're a matrix or list
-                if hasattr(m, 'rows'):
-                    # convert to list of lists
+            if isinstance(m, list):
+                if not strict:
```

Comment:
I think it would be easier to read if it were `if strict: ... else:` instead of `if not strict: ...`

Reply:
ok